core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -765,7 +824,7 @@ public ReplicationPhase(ReplicaRequest replicaRequest, Response finalResponse, S
                     if (shard.currentNodeId().equals(nodes.localNodeId()) == false) {
                         numberOfPendingShardInstances++;
                     }
-                    if (shard.relocating()) {
+                    if (shard.relocating() && shard.relocatingNodeId().equals(nodes.localNodeId()) == false) {
```

Comment:
can we change this block to use the same flow as the doRun (it's equivalent, but I have to double check every time)

```
     if (shard.primary() == false && executeOnReplica == false) {
                    numberOfIgnoredShardInstances++;
                    continue;
                }
                if (shard.unassigned()) {
                    numberOfIgnoredShardInstances++;
                    continue;
                }
                if (nodes.localNodeId().equals(shard.currentNodeId()) == false) {
                    numberOfPendingShardInstances++;
                }
                if (shard.relocating() && shard.relocatingNodeId().equals(nodes.localNodeId()) == false) {
                    numberOfPendingShardInstances++;
                }
```